code/modules/antagonists/mind_flayer/powers/flayer_mobility_powers.dm

```diff
@@ -0,0 +1,219 @@
+//Basically shadow anchor, but the entry and exit point must be computers. I'm not in your walls I'm in your PC
+/datum/spell/flayer/computer_recall
+	name = "Traceroute"
+	desc = "Allows us to cast a mark to a computer. To recall us to this computer, cast this next to a different computer. To check your current mark: Alt click."
+	base_cooldown = 60 SECONDS
+	action_icon_state = "pd_cablehop"
+	upgrade_info = "Halve the time it takes to recharge."
+	power_type = FLAYER_PURCHASABLE_POWER
+	category = FLAYER_CATEGORY_INTRUDER
+	centcom_cancast = FALSE
+	stage = 2
+	base_cost = 125
+	static_upgrade_increase = 25
+	max_level = 2
+	should_recharge_after_cast = FALSE
+	/// The console we currently have a mark on
+	var/obj/marked_computer
+	/// The typecache of things we are allowed to teleport to and from
+	var/list/machine_typecache = list(
```

Comment:
You should have a static list with the possible types, and then a separate empty list to fill the typecache with, I don't really like the way the list gets replaced/reused here

Reply:
We do this in a fair amount of places already though

Reply:
Do we? 
I still think it'd still be ideal to make two lists and have them static.

Reply:
Yeah, I found a lot of things that were either making the list a typecache of itself or it was done on Init and the typecache was added there with written out types.

I can make this list a static one so it won't be remade, or into a global list?

Reply:
Done